lib/utils.js

```diff
@@ -94,19 +94,19 @@ exports.GreatResponse = function (request, options, filterRules) {
             }
 
             if (this.isLeaf) {
-                if (filterRules[this.key] || filterRules[this.parent.key]) {
+                var filter = filterRules.hasOwnProperty(this.key) || filterRules.hasOwnProperty(this.parent.key);
```

Comment:
Do you need to check if it exists and then get it? Could you just get it and see if it exists? If so, it would condense this and the next few lines a bit.